internal/commands/root/sign.go

```diff
@@ -88,18 +88,21 @@ func commandSign(o *options, s *gsio.Streams, args ...string) error {
 		opts.UserName = o.Config.CommitterName
 		opts.UserEmail = o.Config.CommitterEmail
 	}
-	sig, cert, tlog, err := git.Sign(ctx, rekor, userIdent, dataBuf.Bytes(), opts)
+	if o.Config.RekorMode == "offline" {
+		opts.RekorAddr = o.Config.Rekor
```

Comment:
Hmm, I find this a little confusing. If we're "offline", we provide the "RekorAddr"? That's the opposite of what I'd expect.

IIUC this basically does the _same_ signing process, it just additionally stuffs offline verification materials into the git repo.

Reply:
Oh, I see.  If we pass `opts.RekorAddr`, then `signature.Sign` will do the new Rekor log stuff. Otherwise, `git.Sign` will use the `rekor` we pass in.

That feels...a little fraught. I'd rather make the signing mode more explicit in the arguments, or even have separate `git.SignOffline` and `git.SignOnline`.